src/SortChecker.cpp

```diff
@@ -223,6 +255,75 @@ class Visitor : public RecursiveASTVisitor<Visitor> {
   }
 #endif
 
+  // Instrument std::map dtors
+  bool VisitCompoundStmt(CompoundStmt *CS) {
+    auto &SM = Ctx.getSourceManager();
+    auto Loc = CS->getBeginLoc();
+    if (!canInstrument(Loc, SM))
+      return true;
+
+    // Collect names of containers which can be instrumented
+    std::vector<std::string> Maps;
+    auto InsertLoc = CS->getEndLoc();
+    for (auto *S : CS->body()) {
+      if (isa<ReturnStmt>(S)) {
+        InsertLoc = S->getBeginLoc();
+        break;
+      }
```

Comment:
We can have multiple return statements in function + exception throwing

Reply:
It seems that this code doesn't consider that

Reply:
Right, I need to handle inner `CompoundStmt`s and multiple returns, recursively.

Reply:
Anyway, I don't see an easy way to write complete solution which handles returns in nested blocks. So I plan to drop this work.

Reply:
Ok!